Feat/search

hyelin0111 opened the pull request:
1. POST/api/spots/{spotId}/spotLike
 - OK : spot 좋아요 성공, 해당 spot의 좋아요 수 반환
2. DELETE/api/spots/{spotId}/spotLike
 - OK : spot 좋아요 취소 성공, 해당 spot의 좋아요 수 반환

hyelin0111 committed:
feat: 장소 검색 백엔드 구현

1. POST/api/searches : api를 통해 장소 검색 및 db에 저장
	- OK : 장소 검색 성공
2. pom.xml json dependency 추가

hyelin0111 committed:
refactor: 장소 검색 api 코드 수정

1. api결과를 바로 리턴하게 수정
	API 검색 -> 리스트 얻음 -> DB에 존재하지 않는 SPOT 저장 -> SPOT_IMG 저장 -> API 얻은 리스트 반환

2. SearchController -> SpotController 등 파일명 변경
3. 불필요한 코드 삭제

hyelin0111 committed:
Merge branch 'develop' of https://github.com/kvp9356/think-jeonju into
feat/search

# Conflicts:
#	bin/src/main/resources/mybatis-config.xml
#	src/main/java/com/kvp/thinkjeonju/controller/HomeController.java
#	src/main/java/com/kvp/thinkjeonju/exception/common/DataBaseException.java
#	src/main/resources/templates/index.html

hyelin0111 committed:
refactor: bin 폴더 삭제

hyelin0111 committed:
feat: 장소 검색 백엔드, 프론트엔드

1. spot category 추가
2. 장소 검색 미리보기 구현

hyelin0111 committed:
feat: 장소 검색 백엔드, 프론트 엔드

	1. 장소 검색 시 보여지는 리스트에 좋아요 기능 추가
	2. 해당 장소의 좋아요 수 보여주기

hyelin0111 committed:
Merge branch 'develop' into feat/search

mindock commented on src/main/java/com/kvp/thinkjeonju/controller/ApiSpotController.java:
```diff
@@ -0,0 +1,48 @@
+		MemberDTO m = (MemberDTO)session.getAttribute("LOGIN_USER");
+		
+		LikeToDTO like = new LikeToDTO(m.getId(), spotId, 's');
+		spotService.setSpotLike(like);
```
```
spotService.setSpotLike(new LikeToDTO(m.getId(), spotId, 's'));
return new ResponseEntity(spotService.getLikeCnt(spotId), HttpStatus.OK);	
```
이런식으로 쓰는 것이 더 깔끔하지 않을까? 이건 그냥 취향차이여서 물어보는거!

hyelin0111 replied:
훨씬 깔끔하넼ㅋㅋ 이것도 수정했어

vsh123 commented on src/main/java/com/kvp/thinkjeonju/controller/SpotController.java:
```diff
@@ -1,37 +1,53 @@
+			for(int i=0; i<spots.size(); i++) {
+				LikeToDTO like = new LikeToDTO(m.getId(), spots.get(i).getId(), 's');
+				int result = spotService.getIsLike(like);
+				System.out.println(result);
```
TEST 용도였으면 삭제가즈아

hyelin0111 replied:
네~

vsh123 commented on src/main/resources/application.properties:
```diff
@@ -4,4 +4,5 @@ mybatis.mapper-locations=mapper/**/*.xml
-handlebars.expose-session-attributes=true
\ No newline at end of file
+handlebars.expose-session-attributes=true
+handlebars.cache=false
```
이 설정이 intellij에서만 하는게 아니라 sts에서도 해줘야지 html 바로 업데이트 되는거야??(궁금궁금해서 올려요..ㅎ)

hyelin0111 replied:
응 해줘야해ㅎ

vsh123 commented on src/main/resources/templates/index.html:
```diff
@@ -2,6 +2,65 @@
 {{/partial}}
 
 {{#partial "contents"}}
+<style>
```
요기 스타일을 index.css 파일 하나 만들어서 불러오는 건 어떻게 생각해?
<link ~>

그리구 css(style) 삽입 위치는 

{{#partial "head"}}
---요기---
{{/partial}}
로 알고있어요!!

hyelin0111 replied:
파일 다 분리했어요~

vsh123 commented on src/main/resources/templates/index.html:
```diff
@@ -2,6 +2,65 @@
+	</div>
+</div>
+
+<script>
```
이 javascript 파일도 index.js 만들어서 추가하는 건 어때여??
삽입 위치는
{{#partial "script"}}
-----요기!-----
{{/partial}}

vsh123 commented on src/main/resources/templates/spotList.html:
```diff
@@ -0,0 +1,129 @@
+{{/partial}}
+
+{{#partial "contents"}}
+<style>
```
여기두 .css 생성제안이요!

vsh123 commented on src/main/resources/templates/spotList.html:
```diff
@@ -0,0 +1,129 @@
+{{/spots}}
+</div>
+
+<script>
```
여기두 .js 생성제안이요!헿

vsh123 commented on src/main/resources/mapper/search/spot-mapper.xml:
```diff
@@ -1,38 +1,64 @@
+		AND RELATED_ID = #{relatedId}
+	</delete>
+	
+	<select id="getIsLike" resultType="int">
```
getLikeCnt는 resultType이 _int 인데, getIsLike는 int로 설정한 특별한 이유가 있나요??궁금!!

hyelin0111 replied:
특별한 이유는 없고 둘 다 사용가능한 것 같길래

mindock commented on src/main/java/com/kvp/thinkjeonju/controller/SpotController.java:
```diff
@@ -1,37 +1,53 @@
+		
+		if(m != null) {
+			for(int i=0; i<spots.size(); i++) {
+				LikeToDTO like = new LikeToDTO(m.getId(), spots.get(i).getId(), 's');
```
`if(m != null)` 
안의 내용을 service에 넣는것은 어떨까?
setLikeInSpotDTOs와 같은 메소드를 만들어서?

hyelin0111 replied:
수정했어요~

hyelin0111 committed:
refactor: 코드 수정, css/js 파일 분리

	1. index.html css, js 파일 분리
	2. spotList.html css, js 파일 분리
	3. ApiSpotController 코드 수정
	4. SpotController 코드 수정

hyelin0111 committed:
Merge branch 'feat/search' of https://github.com/hyelin0111/think-jeonju
into feat/search

# Conflicts:
#	src/main/java/com/kvp/thinkjeonju/controller/SpotController.java
#	src/main/java/com/kvp/thinkjeonju/repository/SpotMapper.java
#	src/main/java/com/kvp/thinkjeonju/service/SpotService.java

mindock commented on src/main/java/com/kvp/thinkjeonju/service/SpotService.java:
```diff
@@ -232,4 +233,14 @@ public int getIsLike(LikeToDTO like) {
+			if(result == 1) {
+				spots.get(i).setIsLike("true");
+			}
+		}
```
질문!
여기는 spots를 리턴하지 않아도 잘 작동하나요??
like된 것이 저장되나요?

hyelin0111 replied:
오....... 다시 수정할게요

hyelin0111 committed:
refactor: 코드 수정
